src/app/dataset-view/dataset.component.ts

```diff
@@ -51,6 +52,10 @@ export class DatasetComponent
     public isMarkdownEditView = false;
     public markdownText = AppValues.markdownContain;
 
+    private selectedDatasetName: string;
+    private datasetInfo: DatasetInfo = { accountName: "", datasetName: "" };
```

Comment:
I don't think any of these should be component properties, it's easy to confus it with the `datasetBasics`. Instead, just send the `DatasetInfo` , extracted from URL, to GraphQL queries directly.